### PR TITLE
Q3 pre-release Composer install instructions

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) adds examples of downloading Composer metapackages for Magento Commerce minor releases, quality patches, and security patches. It also adds a note that Commerce customers can access these packages two weeks before the General Availability (GA) date.

These instructions were requested by Product Management for use in release enablement.

To make these examples fit in the Upgrade topic, I had to reduce the number of heading sections, which I think needed to be done anyway to simplify the topic.

See MAGEDOC-4042 for more info and links to staging builds.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/install-gde/composer.html
- https://devdocs.magento.com/guides/v2.3/comp-mgr/cli/cli-upgrade.html

whatsnew
Added examples of downloading Composer metapackages for Magento Commerce minor releases, quality patches, and security patches in the [Install](https://devdocs.magento.com/guides/v2.3/install-gde/composer.html) and [CLI Upgrade](https://devdocs.magento.com/guides/v2.3/comp-mgr/cli/cli-upgrade.html) topics.

### CODE REVIEW

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 1]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
-#### Specify Magento packages
+1. Deactivate the {{ ce }} update (for edition upgrade only). _Optional_—If you are upgrading from {{ ce }} to {{ ee }}, deactivate the {{ ce }} update:
 
-Indicate the Magento packages, both the edition (`community` or `enterprise`) and the version (`{{ page.guide_version }}.3`), that you want to upgrade to.
+   ```bash
+   composer remove magento/product-community-edition --no-update
+   ```
 
-_{{ ce }}_:
+1. Indicate the Magento packages, both the edition (`community` or `enterprise`) and the version (`{{ page.guide_version }}.3`), that you want to upgrade to.
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: I don't think we need to change all instances of patch versions for each release. The purpose of the examples is to show users what the command syntax looks like. I think we can reasonably expect users to adapt that syntax to the version they're targeting.

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 2]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
 
-_Magento Commerce_:
+   ```bash
+   composer show magento/product-community-edition {{ page.guide_version }}.* --all | grep -m 1 versions
+   ```
 
-```bash
-composer show magento/product-enterprise-edition {{ page.guide_version }}.* --all | grep -m 1 versions
-```
+   _Magento Commerce_:
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: yes

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 3]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
+   ```
 
-_Magento Open Source_:
+   <div class="bs-callout-tip" markdown="1">
+   To see the full list of available {{ page.guide_version }} version:
 
-```bash
-composer show magento/product-community-edition {{ page.guide_version }}.* --all | grep -m 1 versions
-```
+   _Magento Open Source_:
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: yes

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 4]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
-_{{ ce }}_:
+1. Indicate the Magento packages, both the edition (`community` or `enterprise`) and the version (`{{ page.guide_version }}.3`), that you want to upgrade to.
 
-```bash
-composer require magento/product-community-edition={{ page.guide_version }}.3 --no-update
-```
+   _{{ ce }}_:
 
-_{{ ee }}_:
+   `` `bash
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: done

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 5]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
+   cp composer.json composer.json.bak
+   ```
 
-```bash
-composer remove magento/product-community-edition --no-update
-```
+1. Add or remove specific packages based on your needs.
 
-#### Specify Magento packages
+1. Deactivate the {{ ce }} update (for edition upgrade only). _Optional_—If you are upgrading from {{ ce }} to {{ ee }}, deactivate the {{ ce }} update:
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: Yes and yes.

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 6]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
-    //...
-}
-```
+{% include install/composer-examples.md %}
 
-### Modify the Magento updater
+## (_Optional)_ Modify the Magento updater
 
-_Optional_—If the Magento updater is installed, modify it (it is located in the `update/` directory):
+If the Magento updater is installed, modify it (it is located in the `update/` directory):
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: Based on the existing instructions, it looks like we're asking users to **recreate** a fresh update / directory; not necessarily "modify" it.

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 7]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
+   If you are upgrading from 2.2.x to 2.3.x, remove unused packages with the following command. It is not needed if you are upgrading from 2.3.x.
 
-### Update `autoload`
+   ```bash
+   composer remove --dev sjparkinson/static-review fabpot/php-cs-fixer --no-update
+   ```
 
-Open `composer.json` and edit the `"autoload": "psr-4"` section to include `"Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"`:
+{: #update-autoload }
+1. Open the `composer.json` file and edit the `"autoload"` property to include `"Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"`.
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: thanks for confirming. i'll remove this step.

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 8]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
+   </div>
 
-```bash
-composer require --dev allure-framework/allure-phpunit:~1.2.0 friendsofphp/php-cs-fixer:~2.14.0 lusitanian/oauth:~0.8.10 magento/magento-coding-standard:~3.0.0 magento/magento2-functional-testing-framework:2.4.3 pdepend/pdepend:2.5.2 phpmd/phpmd:@stable phpunit/phpunit:~6.5.0 sebastian/phpcpd:~3.0.0 squizlabs/php_codesniffer:~3.4.0 --sort-packages --no-update
-```
+1. Specify additional packages.
 
-#### Remove unused packages
+   ```bash
+   composer require --dev allure-framework/allure-phpunit:~1.2.0 friendsofphp/php-cs-fixer:~2.14.0 lusitanian/oauth:~0.8.10 magento/magento-coding-standard:~3.0.0 magento/magento2-functional-testing-framework:2.4.3 pdepend/pdepend:2.5.2 phpmd/phpmd:@stable phpunit/phpunit:~6.5.0 sebastian/phpcpd:~3.0.0 squizlabs/php_codesniffer:~3.4.0 --sort-packages --no-update
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: I'm going to err on the side of caution and leave this as-is. It's a bit outside the scope of this PR anyway.

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 9]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
 
-_Optional_—If you are upgrading from {{ ce }} to {{ ee }}, deactivate the {{ ce }} update:
+   ```bash
+   cp composer.json composer.json.bak
+   ```
 
-```bash
-composer remove magento/product-community-edition --no-update
-```
+1. Add or remove specific packages based on your needs.
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: The next step actually provides such an example (upgrading from Open Source to Commerce. I'll consolidate these steps accordingly.

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md [hunk 10]
```diff
@@ -36,93 +36,92 @@ Complete the following prerequisites to prepare your environment before starting
...
 
-<div class="bs-callout-tip" markdown="1">
-To see the full list of available {{ page.guide_version }} version:
+   ```bash
+   composer require magento/product-enterprise-edition={{ page.guide_version }}.3 --no-update
+   ```
 
-_Magento Open Source_:
+   <div class="bs-callout-tip" markdown="1">
+   To see the full list of available {{ page.guide_version }} version:
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: applied locally

File: guides/v2.3/comp-mgr/cli/cli-upgrade.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: checking with @hguthrie about what to do with the reference to this section in the cloud upgrade topic: https://devdocs.magento.com/guides/v2.3/cloud/project/project-upgrade.html#complete-the-upgrade

[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.3/comp-mgr/cli/cli-upgrade.md]: Confirmed; deleting from cloud topic as well.

**Comment by piotrekkaminski** [comment]
approved

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**